### PR TITLE
Added BF16 & FP16 models to PTQ tests

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Added BF16/FP16 models to PTQ tests scope.

### Reason for changes

- Extend e2e with different precisions.

### Related tickets

- 147481

### Tests

- manual post_training_quantization/562/ - passed

### CODE REVIEW

**Comment by KodiaqQ** [comment]
@openvinotoolkit/nncf-maintainers, can someone review this, please. Thank you.

File: tests/post_training/pipelines/base.py [hunk 1]
```diff
@@ -39,6 +39,12 @@
 METRICS_XFAIL_REASON = "metrics_xfail_reason"
 
 
+class PrecisionType(Enum):
```
Comment (AlexanderDokuchaev) [inline tests/post_training/pipelines/base.py:42]: Use torch.dtype directly no need to create yet another enum.

File: tests/post_training/test_quantize_conformance.py [hunk 2]
```diff
@@ -280,6 +281,7 @@ def test_ptq_quantization(
         test_model_param = PTQ_TEST_CASES[test_case_name]
         maybe_skip_test_case(test_model_param, run_fp32_backend, run_torch_cuda_backend, batch_size)
         pipeline_cls = test_model_param["pipeline_cls"]
+        base_precision = test_model_param.get("base_precision", PrecisionType.FP32)
```
Comment (AlexanderDokuchaev) [inline tests/post_training/test_quantize_conformance.py:284]: use `param` to store `base_precision`. 
Like it done for `is_stateful`

https://github.com/openvinotoolkit/nncf/blob/develop/tests/post_training/model_scope.py#L58
https://github.com/openvinotoolkit/nncf/blob/develop/tests/post_training/pipelines/causal_language_model.py#L27

**Review by AlexanderDokuchaev — CHANGES_REQUESTED** [review]

**Comment by AlexanderDokuchaev** [comment]
manual/job/post_training_quantization/563/

**Review by AlexanderDokuchaev — APPROVED** [review]

**Pull request merged**